src/v/cluster/types.h

```diff
@@ -1229,6 +1229,38 @@ enum class partition_operation_type {
 };
 std::ostream& operator<<(std::ostream&, const partition_operation_type&);
 
+/// Notification of topic table state change related to a topic as a whole
+
+enum class topic_table_topic_delta_type {
+    added,
+    removed,
+    properties_updated,
+};
+std::ostream& operator<<(std::ostream&, const topic_table_topic_delta_type&);
+
+struct topic_table_topic_delta {
+    // revision of topic creation command (can be used to identify a topic
+    // incarnation).
+    model::revision_id creation_revision;
```

Comment:
should this be `model::initial_revision` ?

Reply:
This is a bit of a mess, but `model::initial_revision` is used for `metadata.get_remote_revision()` - revision of the topic manifest this topic has been recovered from. So this is different from the revision of the `create_topic` command.